rs/embedders/src/wasm_utils/instrumentation.rs

```diff
@@ -1723,17 +1727,34 @@ fn get_data(
                         "complex initialization expressions for data segments are not supported!".into()
                     ))),
                 },
-
-                _ => return Err(WasmInstrumentationError::WasmDeserializeError(
-                    WasmError::new("no offset found for the data segment".into())
-                )),
+                ic_wasm_transform::DataSegmentKind::Passive => unreachable!(),
             };
 
             Ok((offset, segment.data.to_vec()))
         })
         .collect::<Result<_,_>>()?;
 
-    data_section.clear();
+    // Clear all active data segments, but retain the indices of passive data segments:
+    // * Clear the data of active data segments if (directly or indirectly) followed by a passive segment.
+    // * Delete all active data segments not followed by any passive data segment.
```

Comment:
What's the rationale for this? Why are active data segments treated differently if followed by a passive one? Is the deletion just an optimization?

Reply:
Ah, maybe its so you don't have to re-index the passive data segments and adjust the loads. Correct?

Reply:
Yes, it is only for keeping the indices of the passive data segments. 
The deletion is only for satisfying existing tests and keeping the original behavior for the majority of Wasm binaries that do not contain passive segments.
Probably it makes sense to simplify this and handle all active data segments consistently, regardless of their position, by only erasing their content.